src/Umbraco.Commerce.PaymentProviders.Stripe/StripeCheckoutPaymentProvider.cs

```diff
@@ -224,23 +223,15 @@ public override async Task<PaymentFormResult> GenerateFormAsync(PaymentProviderC
                     { "stripeSessionId", session.Id },
                     { "stripeCustomerId", session.CustomerId }
                 },
-                Form = new PaymentForm(ctx.Urls.ContinueUrl, PaymentFormMethod.Post)
+                Form = new PaymentForm(ctx.Urls.ErrorUrl, PaymentFormMethod.Post)
                     .WithAttribute("onsubmit", "return handleStripeCheckout(event)")
-                    .WithJsFile("https://js.stripe.com/v3/")
                     .WithJs(@"
-                        var stripe = Stripe('" + publicKey + @"');
                         window.handleStripeCheckout = function (e) {
                             e.preventDefault();
-                            stripe.redirectToCheckout({
-                                sessionId: '" + session.Id + @"'
-                            }).then(function (result) {
-                              // If `redirectToCheckout` fails due to a browser or network
-                              // error, display the localized error message to your customer
-                              // using `result.error.message`.
-                            });
+                            window.location.href = '" + session.Url + @"';
```

Comment:
If we have a URL in the session object, we might be able to do away with this javascript entirely and just set the PaymentForms URL to the session URL and it's method to GET. This would avoid ever redirecting to anything other than the session URL whether javascript is enabled or not.

Reply:
that's my first thought. Too bad that when the form action is GET, Commerce modifies the form's url which lead to an invalid url. Hence, that script.

Reply:
Modifies it in what way?

Reply:
Oh wait, I see what we do, we strip the query string parameter and make them form inputs. Does this not work?

Reply:
the whole part after the hashtag(#) in the url is removed

Reply:
Ahh ok, what does a session URL look like?

Reply:
`https://checkout.stripe.com/c/pay/cs_test_a1O9YfXImqGp3AGI3BnVtfB2XhDMuT55VHlmVPllPewm1jCA9yXtJO3IeQ#fidpamZkaWAnPydgaycpJ2R1bE5gfCc%2FJ3VuWnFgdnFaMDRKYm92S01gT0JGXWZIPF1HXFRVQHZxbTVzRHMxVmhnN0FcRjdObU5KdURhRFNtYnw9Z3IzX219dm5ifVVocUhtdWlPfzQ1fEF%2FSjJQNHZrSzxfZDFgRDc1NVdnQVIwUU9KJyknY3dqaFZgd3Ngdyc%2FcXdwYCknaWR8anBxUXx1YCc%2FJ3Zsa2JpYFpscWBoJyknYGtkZ2lgVWlkZmBtamlhYHd2Jz9xd3BgeCUl`

Reply:
Actually, it looks like the Stripe docs give an example

```
https://checkout.stripe.com/c/pay/cs_test_a11YYufWQzNY63zpQ6QSNRQhkUpVph4WRmzW0zWJO2znZKdVujZ0N0S22u#fidkdWxOYHwnPyd1blpxYHZxWjA0SDdPUW5JbmFMck1wMmx9N2BLZjFEfGRUNWhqTmJ%2FM2F8bUA2SDRySkFdUV81T1BSV0YxcWJcTUJcYW5rSzN3dzBLPUE0TzRKTTxzNFBjPWZEX1NKSkxpNTVjRjN8VHE0YicpJ2N3amhWYHdzYHcnP3F3cGApJ2lkfGpwcVF8dWAnPyd2bGtiaWBabHFgaCcpJ2BrZGdpYFVpZGZgbWppYWB3dic%2FcXdwYHgl
```

So yea, I can see we need to include the # fragement.

In that case, lets go with your solution and we can do some testing in v15 to see if we pass the hash fragment in a form action URL, will that get submitted to stripe and work.

Thanks for clarifying the issue 👍